client/src/Monomino.ts

```diff
@@ -0,0 +1,53 @@
+import Phaser, { Scene } from "phaser";
+
+import { TILE_SIZE, TILE_SCALE } from "common/shared";
+import { TetrominoType } from "common/TetrominoType";
+
+export class Monomino {
+    public position: [number, number];
+
+    private type: TetrominoType;
+    private ownerId!: 0 | 1 | 2 | 3 | null;
+    private sprite?: Phaser.GameObjects.Image;
+
+    constructor(
+        type: TetrominoType,
+        position: [number, number],
+        ownerId: 0 | 1 | 2 | 3 | null
+    ) {
+        this.type = type;
+        this.position = position;
+        this.setOwnerId(ownerId);
+    }
+
+    public destroy() {
+        if (this.sprite) {
+            this.sprite.destroy();
+        }
+    }
+
+    public getCopy(): Monomino {
+        return new Monomino(this.type, this.position, this.ownerId);
```

Comment:
In `line clearing` branch I had to change `this.position` to `this.position.slice()` because the `dropSprites()`, now with line clearing, won't be enough to just shallow clone the position, as line clearing will move the positions of monominoes on the board. 

This change is already in `line-clearing` branch, but should it be here?

(If the position is shallow cloned, `currentTetromino` will have its monomino referencing the same position array with the monominoes on the board. So when a line is cleared and the monomino position shifted, the currentTetromino will have two monominoes overlap and will look like it lost a monomino somehow.)

Reply:
Resolved in commit [10979e6](https://github.com/ECE493G14Capstone/Capstone/pull/77/commits/10979e6300d1b44e80ed69774921f0ef5f88099d).